### ISSUE TITLE
Deprecate the `watch` command

### ISSUE THREAD

**Issue by eggplantzzz** [trufflesuite/truffle#3227]

**Comment by eliotstock** [comment]
`truffle watch` says 'See https://github.com/trufflesuite/truffle/issues/3227 for more information.' So, where's the more information?

Also, even in version 5.3.1, `truffle watch` simply exits.

**Comment by eggplantzzz** [comment]
@eliotstock So this was meant to be an issue to track notes about the deprecation of the `watch` command. So far there has not really been any discussion. Looks like we introduced a bug at some point...

**Comment by rakeshta** [comment]
Why is `watch` command being deprecated?

**Comment by eggplantzzz** [comment]
Hey @rakeshta, we think that a task running utility like gulp or grunt can more appropriately and robustly handle this kind of thing.

**Comment by rakeshta** [comment]
@eggplantzzz that's true. But if you look at comparable tools like `tsc`, `mocha` or `jest`, all of them have the `watch` feature.
Sometimes, you just want the convenience without having to install and configure more tools.
